src/client/sandbox/event/message.js

```diff
@@ -172,12 +172,19 @@ export default class MessageSandbox extends SandboxBase {
         if (!this._isIframeRemoved() && (isIframeWithoutSrc || !isCrossDomainWindows(targetWindow, this.window) &&
                                                                targetWindow[this.RECEIVE_MSG_FN])) {
             //NOTE: postMessage delay imitation
-            nativeMethods.setTimeout.call(this.topWindow, () =>
+            nativeMethods.setTimeout.call(this.topWindow, () => {
+                // NOTE: We leave the capability to communicate with a removed frame. Unfortunately, this
+                // cannot be done in Safari, because it restricts operations with removed windows (GH-171).
+                var isIframeRemoved = !this.window.top;
```

Comment:
It's should be `isRemovedIframe`. Anyway, `removed` is not the best word here, `detached` is better.